### ISSUE TITLE
Question regarding computing EER

### ISSUE THREAD

**Issue by jlian2** [clovaai/voxceleb_trainer#11]
This your method to compute EER:
![image](https://user-images.githubusercontent.com/61141271/80098409-d36db080-853a-11ea-94ff-5cfee6e56764.png)
I have made a test and I got:
![image](https://user-images.githubusercontent.com/61141271/80098447-e5e7ea00-853a-11ea-90f1-9752f7336ffd.png)
,that is to say, eer is 0.4
However, there is another version to compute EER:(https://github.com/nesl/asvspoof2019/blob/master/eval_metrics.py), I got 0.2:
![image](https://user-images.githubusercontent.com/61141271/80098507-03b54f00-853b-11ea-8c36-9766a683751a.png)
![image](https://user-images.githubusercontent.com/61141271/80098532-0c0d8a00-853b-11ea-8658-9819ead13b76.png)
I find that you take the maximum value of fpr[index] and fnr[index] rather than the mean. Why for this? Plus, even if I take the mean value, the result is still not consistent.

**Comment by 009deep** [comment]
@jlian2 you  have used t_score, n_score as input to second function vs scores , labels in first one. There are multiple EER evaluation. My personal favorite is [this one](https://yangcha.github.io/EER-ROC/). Having said that it doesn't matter which implementation is used for same input output is always more or less same with few decimal difference.

**Comment by jlian2** [comment]
@009deep Thanks! But as you can see in the figure for that example, one eer is 0.4 while another is 0.2. Even if i take the mean 0.3, there is still a relatively big difference.

**Comment by 009deep** [comment]
@jlian2 Give same input to both functions. Currently you are passing different inputs.

**Comment by joonson** [comment]
Could there be difference in the way the scripts deal with equal scores (which would not happen with real examples)? What happens if you add small random number to all of the scores.

**Comment by joonson** [comment]
Could there be difference in the way the scripts deal with equal scores (which would not happen with real examples)? What happens if you add small random number to all of the scores.

**Comment by joonson** [comment]
Could there be difference in the way the scripts deal with equal scores (which would not happen with real examples)? What happens if you add small random number to all of the scores.

**Comment by jlian2** [comment]
@joonson The results indicate: yes, using random input.

**Comment by joonson** [comment]
@jlian2 Do you mean the results are the same, if you give the scores a small random offset?

**Comment by jlian2** [comment]
@joonson Sorry about that, I mean given the same input, the result is different. As you can see in the figure: eer is 0.4 and eer is 0.2

**Comment by joonson** [comment]
I think that the effect on EER might be over-exaggerated because of the small number of samples, when it is only a difference of one sample. Does it make any difference over real data (e.g. VoxCeleb test set using the pre-trained model)? Thanks.

**Comment by jlian2** [comment]
@joonson That may make sense. I am currently try on other datasets. And in your implementation of EER, why not use this:
![image](https://user-images.githubusercontent.com/61141271/80116984-9d3d2a80-8554-11ea-8878-c1e344009cf8.png)
Because, mean value typically lead to a smaller value. And When I am trying another dataset, I got 21 for max and 19 for mean.

**Comment by joonson** [comment]
Yes, this might make a very small improvement in EER. I don't think there is one _correct_ way to deal with this scenario in the definition of EER, but here I chose to take a conservative measure.
